src/opus.c

```diff
@@ -30,23 +31,30 @@
 #endif
 
 #include "opus.h"
+#include "celt/mathops.h"
 #include "opus_private.h"
 
 #ifndef DISABLE_FLOAT_API
-OPUS_EXPORT void opus_pcm_soft_clip(float *_x, int N, int C, float *declip_mem)
+
+static void opus_pcm_soft_clip_impl(float *_x, int N, int C, float *declip_mem, int use_arch, int arch)
```

Comment:
Why do you need both use_arch and arch, instead of just setting arch=0 to get the C version?

Reply:
I removed use_arch and adjusted the relevant parts.
Arch is always passed and when it's 0, it is going to yield the C implementation.